### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by karots123** [microsoft/DeepSpeed#4274]
Hello，we meet an issue during using deepspeed pipeline engine. When we set the activation_checkpoint_interval=0 in the PipelineModule class, the structure works. But when we set it >0, the system will report

**Comment by karots123** [comment]
What an interesting bug！
This bug was raised by https://github.com/microsoft/DeepSpeed/issues/688 before. 
The core is the ‘_is_checkpointable‘ function which hopes to filter embedding layer and loss calculation layer instead of choosing them as the checkpointable layer.
However, if we don't name our model class with 'ParallelTransformerLayerPipe' and define a new class named 'GPT2ModelPipe' which inherits 'PipelineModule',  the _is_checkpointable func will choose 
"params = [f.parameters() for f in funcs if isinstance(f, torch.nn.Module)]"
 "return any(len(list(p)) > 0 for p in params)" 
as output, which always let embeding layer as the checkpoint layer.
 Similar to the auther ot issue 688, we 'd be willing to put in some work to fix this bug, as we'd love to be able to run the  deepspeed library without maintaining our fork!

**Comment by tohtana** [comment]
> https://github.com/microsoft/DeepSpeed/commit/f16f718bf8f517e7418ed95feaa762992e493d0c has fixed this bug

Hello @karots123, do you mean the latest version does not have this issue?

**Comment by phamkhactu** [comment]
Hello @karots123 @tohtana 
I encounter the same problem with version 0.10.3. 
Do you have any information?

**Comment by jiashu-z** [comment]
Hi @karots123 @tohtana
I encountered this issue with version 0.12.2.

**Comment by brunomaga** [comment]
Im using `deepspeed==0.12.6` and that issue is still there. The issue in my case was that I was doing a transpose of the final logits with `torch.swapaxes(logits,1,2)`. This was then forced to the loss module. I fixed it by forcing loss to compute gradients with:

```
outputs = model(inputs)
outputs = outputs.requires_grad_(True)   # <----- added this
loss = criterion(output, logits)
```

**Comment by lyriccoder** [comment]
the same issue with deepspeed==0.12.3 and 0.16.2. Do u know version which works well?

**Comment by XU-YIJIE** [comment]
> the same issue with deepspeed==0.12.3 and 0.16.2. Do u know version which works well?

encountered same issue with 0.16.2, solved by downgrading to 0.15.1
